libcaliptra/src/caliptra_api.c

```diff
@@ -127,6 +127,8 @@ static int caliptra_mailbox_write_fifo(struct caliptra_buffer *buffer)
 
     // Write DLEN
     caliptra_mbox_write_dlen(buffer->len);
+    if (buffer->data == NULL)
```

Comment:
Should this be up a few lines and return non-zero?

Reply:
This covers the case for a request that does not have a payload.